src/algebra/order/lattice_group.lean

```diff
@@ -72,17 +72,22 @@ variables {α : Type u} [lattice α] [comm_group α]
 @[to_additive]
 lemma mul_sup [covariant_class α α (*) (≤)] (a b c : α) : c * (a ⊔ b) = (c * a) ⊔ (c * b) :=
 begin
-  refine le_antisymm _ (by simp),
+  refine le_antisymm _ (by
+  { rw [sup_le_iff, mul_le_mul_iff_left, mul_le_mul_iff_left, and_iff_left],
+    exact le_sup_left, exact le_sup_right, }),
```

Comment:
Nesting `by` blocks isn't very idiomatic:
```suggestion
  refine le_antisymm _ _,
  { rw [sup_le_iff, mul_le_mul_iff_left, mul_le_mul_iff_left, and_iff_left],
    exact le_sup_left, exact le_sup_right, },
```

Reply:
That didn't seem to work.

Reply:
How about something like this?

https://github.com/leanprover-community/mathlib/pull/18046/commits/fb322ef50af13a466156274e9e72bb6dbe5083b3

Reply:
Slight improvement https://github.com/leanprover-community/mathlib/pull/18046/commits/1bae2cb37a1954022b8fecedce7fc09393ec4d48

Reply:
Oh sorry, that didn't work because the goals changed order
``` 
  refine le_antisymm _ _,
  swap
  { rw [sup_le_iff, mul_le_mul_iff_left, mul_le_mul_iff_left, and_iff_left],
    exact le_sup_left, exact le_sup_right, },
```
would work, as would just swapping around the goals in the proof text